### PR TITLE
Yagna uninstall instructions

### PR DESCRIPTION
**Pull request by jalas167**

### CODE REVIEW

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
+---
+title: How to remove Yagna?
+description: Step-by-step guide to removing Yagna components, configs, and data for Requestors.
+type: noindex
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:4]: Noindex should probably not be here. We want that to be searchable

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+---
+
+# Golem Requestor uninstallation guide
+
+{% alert level="warning" %}
+Before removing Yagna please make sure you have backup-up your private key to the address where you have GLMs. You can find instructions [here](/docs/creators/javascript/examples/tools/golem-wallet-backup).
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:10]: Before removing Yagna, please make sure **you have backed up** your private key.
```suggestion
Before removing Yagna, please make sure you have backed up your private key. You can find the instructions [here](/docs/creators/javascript/examples/tools/golem-wallet-backup).
```

Backing up to a place where you have GLM's doesn't make sense.

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+{% /alert %}
+
+## Quick path (Linux only)
+
+If you used default settings for installation, this set of commands should be sufficient:
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:15]:
```suggestion
If you installed Yagna using our quick-installer, execute the following commands to remove all associated components:
```

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+```bash
+rm ~/.local/bin/yagna
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:18]: Probably also need to uninstall GFTP

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+rm -fr ~/.local/share/ya-installer
+```
+
+## Detailed instructions for custom installations
+
+### Removing Yagna binaries
+
+{% tabs %}
+{% tab label="Linux" %}
+The default Yagna binaries (actually link to binaries) location is: `~/.local/bin/yagna`.
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:29]:
```suggestion
Yagna binaries are typically located at: `~/.local/bin/yagna`.
```

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+```bash
+rm ~/.local/bin/yagna
+```
+
+{% /tab %}
+{% tab label="Windows" %}
+The Yagna is "installed" manually by unziping the archive file. All you need to do is find and remove two files: `yagna.exe` and `gftp.exe`.
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:37]:
```suggestion
Yagna is manually installed by extracting from an archive. Simply locate and delete the `yagna.exe` and `gftp.exe` files.
```

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+If they are within your path, you can find them using:
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:38]:
```suggestion
If they are in your system's PATH, you can locate them with the following command:
```

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+
+{% /tab %}
+{% /tabs %}
+
+### Removing Yagna data
+
+{% tabs %}
+{% tab label="Linux" %}
+
+Yagna daemon configuration and persistent files are stored in the location defined by the `YAGNA_DATADIR` environment variable. If it was not set, then the default location for Yagna data is: `~/.local/share/yagna`.
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:54]:
```suggestion
The configuration and data files for the Yagna daemon are saved in a directory specified by the `YAGNA_DATADIR` environment variable. If you haven't set this variable, Yagna uses a default directory: `~/.local/share/yagna`.


```

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+To check if you use a custom data location run:
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:56]:
```suggestion
To find out if your data is stored in a custom location, type this command in the terminal:
```

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+
+```bash
+echo $YAGNA_DATADIR
+```
+
+To remove Yagna data in default location use:
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:62]:
```suggestion
If your Yagna data is in the default directory and you wish to remove it, use the following command:
```

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+
+```bash
+rm -r ~/.local/share/yagna
+```
+
+{% /tab %}
+{% tab label="Windows" %}
+Yagna daemon configuration and persistent files are stored in the location defined by the `YAGNA_DATADIR` environment variable. If it is not set, then the default location for Yagna data is: `%HOMEPATH%\AppData\Roaming\GolemFactory\yagna`.
+
+To check if you use a custom data location run:
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:72]:
```suggestion
To verify whether you're using a custom location for Yagna data, execute this command:
```

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+
+```sh
+set YAGNA_DATADIR
+```
+
+To remove Yagna data in default location use:
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:78]:
```suggestion
If your Yagna data is in the default directory and you wish to remove it, use the following command:
```

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+
+{% /tab %}
+{% /tabs %}
+
+### Removing Yagna installer files
+
+{% tabs %}
+{% tab label="Linux" %}
+
+To remove Yagna installer files in default location use:
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:92]:
```suggestion
To delete the Yagna installer files from their default location, enter the following command:
```

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 14]
```diff
@@ -0,0 +1,124 @@
...
+{% tab label="Windows" %}
+
+On Windows, you can remove the zip file downloaded from Github.
+
+{% /tab %}
+{% /tabs %}
+
+## Uninstalling `gvmkit-build`
+
+You can check if you have installed `gvmkit-build` globally using:
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:108]:
```suggestion
To determine if `gvmkit-build` is installed globally on your system, run the following command:
```

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 15]
```diff
@@ -0,0 +1,124 @@
...
+
+```sh
+npm ls -g | grep gvmkit
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:111]: Does this work on windows too? Also this doesn't really cover installations using pip

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 16]
```diff
@@ -0,0 +1,124 @@
...
+```
+
+If so, you can uninstall it with the following command:
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:114]:
```suggestion
If the command finds `gvmkit-build`, you can remove it by executing:
```

File: src/pages/docs/providers/provider-uninstall.md [hunk 17]
```diff
@@ -0,0 +1,99 @@
+---
+title: How to remove Yagna?
+description: Step-by-step guide to removing Yagna components, configs, and data for Providers.
+type: noindex
+---
+
+# Golem Provider uninstallation guide
+
+{% alert level="warning" %}
+Before removing Yagna please make sure you have backup-up your private key to the address where you received GLMs. You can find instructions [here](/docs/providers/wallet/backup).
```
Comment (cryptobench) [inline src/pages/docs/providers/provider-uninstall.md:10]:
```suggestion
Before removing Yagna, please make sure you have backed up your private key. You can find the instructions [here](/docs/providers/wallet/backup).
```

File: src/pages/docs/providers/provider-uninstall.md [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+{% /alert %}
+
+## Quick path
+
+If you used default settings for installation, this set of commands should be sufficient:
```
Comment (cryptobench) [inline src/pages/docs/providers/provider-uninstall.md:15]:
```suggestion
If you installed the provider using our quick-installer, execute the following commands to remove all associated components:
```

File: src/pages/docs/providers/provider-uninstall.md [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+
+```bash
+rm ~/.local/bin/yagna
+rm ~/.local/bin/golemsp
+rm ~/.local/bin/ya-provider
+rm -fr ~/.local/share/yagna
```
Comment (cryptobench) [inline src/pages/docs/providers/provider-uninstall.md:21]:
```suggestion
rm -rf ~/.local/share/yagna
```

File: src/pages/docs/providers/provider-uninstall.md [hunk 20]
```diff
@@ -0,0 +1,99 @@
...
+rm -fr ~/.local/share/ya-installer
```
Comment (cryptobench) [inline src/pages/docs/providers/provider-uninstall.md:22]:
```suggestion
rm -rf ~/.local/share/ya-installer
```

File: src/pages/docs/providers/provider-uninstall.md [hunk 21]
```diff
@@ -0,0 +1,99 @@
...
+rm -fr ~/.local/share/ya-provider
```
Comment (cryptobench) [inline src/pages/docs/providers/provider-uninstall.md:23]:
```suggestion
rm -rf ~/.local/share/ya-provider
```

File: src/pages/docs/providers/provider-uninstall.md [hunk 22]
```diff
@@ -0,0 +1,99 @@
...
+rm -fr ~/.local/lib/yagna
```
Comment (cryptobench) [inline src/pages/docs/providers/provider-uninstall.md:24]:
```suggestion
rm -rf ~/.local/lib/yagna
```

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 23]
```diff
@@ -0,0 +1,124 @@
...
+Before removing Yagna please make sure you have backup-up your private key to the address where you have GLMs. You can find instructions [here](/docs/creators/javascript/examples/tools/golem-wallet-backup).
+{% /alert %}
+
+## Quick path (Linux only)
+
+If you used default settings for installation, this set of commands should be sufficient:
+
+```bash
+rm ~/.local/bin/yagna
+rm -fr ~/.local/share/yagna
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:19]:
```suggestion
rm -rf ~/.local/share/yagna
```

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 24]
```diff
@@ -0,0 +1,124 @@
...
+rm -fr ~/.local/share/ya-installer
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:20]:
```suggestion
rm -rf ~/.local/share/ya-installer
```

File: src/pages/docs/providers/provider-uninstall.md [hunk 25]
```diff
@@ -0,0 +1,99 @@
...
+```
+
+## Detailed instructions for custom installations
+
+### Removing binaries
+
+The default Yagna binaries (actually link to binaries) location is: `~/.local/bin/yagna`.
+
+```bash
+rm ~/.local/bin/yagna
```
Comment (cryptobench) [inline src/pages/docs/providers/provider-uninstall.md:34]:
```suggestion
rm ~/.local/bin/yagna && rm ~/.local/bin/golemsp && rm ~/.local/bin/ya-provider
```

Let's just make it simple with a one-liner

File: src/pages/docs/providers/provider-uninstall.md [hunk 26]
```diff
@@ -0,0 +1,99 @@
...
+rm ~/.local/bin/golemsp
```
Comment (cryptobench) [inline src/pages/docs/providers/provider-uninstall.md:35]:
```suggestion

```

**Review by cryptobench — CHANGES_REQUESTED** [review]
@jalas167 I think that the article could benefit from a "Quick uninstall" at the top which just has a single command like `rm ... && rm -rf ... && rm ....`. and so on so that you don't need to skim this whole article. A simple copy and paste and its done.

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md (hunk 1 above)
Comment (cryptobench) [reply src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:4]: I will edit that as it doesn't allow me to search on the preview. I think we should always check how our search looks when adding new articles. They should be easy to find using proper keywords.

Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:4]:
```suggestion
type: article
```

**Review by cryptobench — CHANGES_REQUESTED** [review]

**Comment by cryptobench** [comment]
@jalas167 

image


I would like to see titles like `How to uninstall the Golem Provider` and `How to uninstall the Golem Requestor`. You can probably use that as main titles of the articles. I don't believe our current titles are sufficient when searching.

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md (hunk 15 above)
Comment (jalas167) [reply src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:119]: fixed

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md (hunk 4 above)
Comment (jalas167) [reply src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:18]: added in next commit

File: src/pages/docs/providers/provider-uninstall.md [hunk 27]
```diff
@@ -0,0 +1,97 @@
...
+
+### Removing binaries
+
+Yagna binaries are typically located at: `~/.local/bin/`. Run the following command to remove all of the components.
+
+```bash
+rm ~/.local/bin/yagna && rm ~/.local/bin/golemsp && rm ~/.local/bin/ya-provider
+```
+
+### Removing Yagna binaries
```
Comment (cryptobench) [inline src/pages/docs/providers/provider-uninstall.md:37]: The title is incorrect. Its not the binaries in the datadir.

File: src/pages/docs/providers/provider-uninstall.md [hunk 28]
```diff
@@ -0,0 +1,97 @@
...
+
+To remove the data in the default location, execute the command:
+
+```bash
+rm -r ~/.local/share/yagna
+```
+
+## Removing Yagna installer files
+
+To remove the data in the default location, execute the command:
```
Comment (cryptobench) [inline src/pages/docs/providers/provider-uninstall.md:55]: To me it's not clear what these files cover. I don't know whats in the directory and what its used for

File: src/pages/docs/providers/provider-uninstall.md [hunk 29]
```diff
@@ -0,0 +1,97 @@
...
+
+```bash
+rm -r ~/.local/share/ya-provider
+```
+
+## Removing the Runtimes directory
+
+The runtimes directory is defined by the `$EXE_UNIT_PATH` environment variable. If this variable is not defined the default location is: `~/.local/lib/yagna`.
```
Comment (cryptobench) [inline src/pages/docs/providers/provider-uninstall.md:79]:
```suggestion
The runtimes directory is defined by the `EXE_UNIT_PATH` environment variable. If this variable is not defined the default location is: `~/.local/lib/yagna`.
```

File: src/pages/docs/providers/provider-uninstall.md [hunk 30]
```diff
@@ -0,0 +1,97 @@
...
+
+To remove the data in the default location, execute the command:
+
+```bash
+rm -r ~/.local/lib/yagna
+```
+
+{% docnavigation title="See also" %}
+
+- If you are looking for intructions for Requestors see [here](/docs/creators/javascript/examples/tools/requestor-uninstall).
```
Comment (cryptobench) [inline src/pages/docs/providers/provider-uninstall.md:95]:
```suggestion
- If you're looking for intructions on how to uninstall the requestor, [click here](/docs/creators/javascript/examples/tools/requestor-uninstall).
```

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 31]
```diff
@@ -0,0 +1,181 @@
+---
+title: How to uninstall Golem Requestor?
+description: Step-by-step guide to removing Yagna components, configs, and data for Requestors.
+type: Instructions
+---
+
+# Golem Requestor uninstallation guide
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:7]:
```suggestion
# How to Uninstall the Golem Requestor
```

File: src/pages/docs/providers/provider-uninstall.md [hunk 32]
```diff
@@ -0,0 +1,97 @@
+---
+title: How to uninstall Golem Provider?
+description: Step-by-step guide to removing Yagna components, configs, and data for Providers.
+type: Instructions
+---
+
+# Golem Provider uninstallation guide
```
Comment (cryptobench) [inline src/pages/docs/providers/provider-uninstall.md:7]:
```suggestion
# How to uninstall the Golem Provider
```

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 33]
```diff
@@ -0,0 +1,181 @@
+---
+title: How to uninstall Golem Requestor?
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:2]:
```suggestion
title: How to uninstall the Golem Requestor
```

File: src/pages/docs/providers/provider-uninstall.md [hunk 34]
```diff
@@ -0,0 +1,97 @@
+---
+title: How to uninstall Golem Provider?
```
Comment (cryptobench) [inline src/pages/docs/providers/provider-uninstall.md:2]:
```suggestion
title: How to uninstall the Golem Provider
```

**Review by cryptobench — CHANGES_REQUESTED** [review]

File: src/pages/docs/providers/provider-uninstall.md (hunk 28 above)
Comment (jalas167) [reply src/pages/docs/providers/provider-uninstall.md:55]: binaries are there

File: src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md [hunk 35]
```diff
@@ -0,0 +1,181 @@
...
+
+{% /tab %}
+{% tab label="Windows" %}
+
+On Windows, you can remove the zip file downloaded from Github.
+
+{% /tab %}
+{% /tabs %}
+
+## Uninstalling `gvmkit-build`
```
Comment (cryptobench) [inline src/pages/docs/creators/javascript/examples/tools/requestor-uninstall.md:107]:
```suggestion
## How to uninstall gvmkit-build
```

**Review by cryptobench — CHANGES_REQUESTED** [review]

**Review by cryptobench — APPROVED** [review]

**Pull request merged**